web3/__init__.py

```diff
@@ -1,9 +1,16 @@
 import pkg_resources
 import sys
+import warning
 
 if sys.version_info < (3, 5):
     raise EnvironmentError("Python 3.5 or above is required")
 
+if sys.version_info < (3, 6):
+    warnings.warn(
+        "Support for Python 3.5 will be removed in web3.py v5",
+        category=DeprecationWarning,
+        stacklevel=2)
+
```

Comment:
I opted to have both warnings print in the case that the python version is less than 3.5, so that the user doesnt have to update to 3.5 to learn that support for 3.5 will be dropped soon.

Reply:
The first one is an exception and will exit the python process before hitting this new warning.... so I don't think this behaves as you expect.